sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -80,7 +80,7 @@ int coap_parse(coap_pkt_t *pkt, uint8_t *buf, size_t len)
         return -EBADMSG;
     }
 
-    if (coap_get_token_len(pkt) > COAP_TOKEN_LENGTH_MAX) {
+    if ((pkt->hdr->ver_t_tkl & 0xf) == 0xf) {
```

Comment:
Since the only way for `tkl` to be invalid is being `15`, the `DEBUG(..)` can be more precise, e.g. `DEBUG("nanocoap: token length is reserved value 15, invalid for extended token length field.")`.
(I personally would switch `0xf` with `15`, sticking to the RFC :monkey:)